fastly/resource_fastly_service_v1_splunk_test.go

```diff
@@ -0,0 +1,404 @@
+		ResponseCondition: "error_response_5XX",
+	}
+
+	splunkLogTwo := gofastly.Splunk{
+		Name:              "test-splunk-2",
+		URL:               "https://mysplunkendpoint.example.com/services/collector/event",
+		Token:             "test-token",
+		Format:            "%h %l %u %{now}V %{req.method}V %{req.url}V %>s %{resp.http.Content-Length}V",
+		FormatVersion:     2,
+		Placement:         "waf_debug",
+		ResponseCondition: "ok_response_2XX",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1SplunkConfig_complete(serviceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1SplunkAttributes(&service, []*gofastly.Splunk{&splunkLogOne}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", serviceName),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "splunk.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceV1SplunkConfig_update(serviceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1SplunkAttributes(&service, []*gofastly.Splunk{&splunkLogOneUpdated, &splunkLogTwo}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", serviceName),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "splunk.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFastlyServiceV1_splunk_default(t *testing.T) {
+	var service gofastly.ServiceDetail
+	serviceName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	splunkLog := gofastly.Splunk{
+		Name:          "test-splunk",
+		URL:           "https://mysplunkendpoint.example.com/services/collector/event",
+		Token:         "test-token",
+		Format:        "%h %l %u %t \"%r\" %>s %b",
+		FormatVersion: 2,
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1SplunkConfig_default(serviceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1SplunkAttributes(&service, []*gofastly.Splunk{&splunkLog}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", serviceName),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "splunk.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccFastlyServiceV1_splunk_env(t *testing.T) {
+	var service gofastly.ServiceDetail
+	serviceName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	// set env variable to something we expect
+	resetEnv := setSplunkEnv("test-token", t)
```

Comment:
I am a bit confused about this line.  Why do we only set this here?  I can see in the other tests that the value 'test-token' is being used by it is only set and unset here.

Reply:
We only invoke `setSplunkEnv` once because we have only one test to make sure that the provider applies the `FASTLY_SPLUNK_TOKEN` environment variable. At the end of the test, we delete the `FASTLY_SPLUNK_TOKEN` test variable and reset it with the existing value. This test suite uses the exact same pattern to test the `schema.EnvDefaultFunc` functionality as all other suites of this provider.